automation/system_test/prepare.py

```diff
@@ -326,49 +369,77 @@ def main():
+    default=None,
+    help="Override default mlrun docker image registry.",
+)
+@click.option(
+    "--override-image-repo",
+    "-mrep",
+    default=None,
+    help="Override default mlrun docker image repository name.",
+)
+@click.option(
+    "--override-mlrun-images",
+    "-omi",
+    default=None,
+    help="Override default images (comma delimited list).",
 )
 @click.argument("data-cluster-ip", type=str, required=True)
 @click.argument("data-cluster-ssh-password", type=str, required=True)
 @click.argument("app-cluster-ssh-password", type=str, required=True)
-@click.argument("github-access-key", type=str, required=True)
+@click.argument("github-access-token", type=str, required=True)
 @click.argument("mlrun-dbpath", type=str, required=True)
-@click.argument("v3io-api", type=str, required=True)
-@click.argument("v3io-username", type=str, required=True)
-@click.argument("v3io-access-key", type=str, required=True)
-@click.argument("v3io-password", type=str, default=None, required=False)
+@click.argument("webapi-direct-url", type=str, required=True)
+@click.argument("username", type=str, required=True)
+@click.argument("access-key", type=str, required=True)
+@click.argument("password", type=str, default=None, required=False)
+@click.option(
+    "--debug",
+    "-d",
+    is_flag=True,
+    help="Don't run the ci only show the commands that will be run",
+)
 def run(
     mlrun_version: str,
-    mlrun_repo: str,
+    override_image_registry: str,
+    override_image_repo: str,
+    override_mlrun_images: str,
     data_cluster_ip: str,
     data_cluster_ssh_password: str,
     app_cluster_ssh_password: str,
-    github_access_key: str,
+    github_access_token: str,
     mlrun_dbpath: str,
-    v3io_api: str,
-    v3io_username: str,
-    v3io_password: str,
-    v3io_access_key: str,
+    webapi_direct_url: str,
+    username: str,
+    access_key: str,
+    password: str,
+    debug: bool,
 ):
-    system_test_ci_runner = SystemTestCIRunner(
+    system_test_preparer = SystemTestPreparer(
         mlrun_version,
-        mlrun_repo,
+        override_image_registry,
+        override_image_repo,
+        override_mlrun_images,
         data_cluster_ip,
         data_cluster_ssh_password,
         app_cluster_ssh_password,
-        github_access_key,
+        github_access_token,
         mlrun_dbpath,
-        v3io_api,
-        v3io_username,
-        v3io_password,
-        v3io_access_key,
+        webapi_direct_url,
+        username,
+        access_key,
+        password,
+        debug,
     )
     try:
-        system_test_ci_runner.run()
-    except Exception as e:
-        logger.error("Failed running system test automation", exception=e)
-    finally:
-        system_test_ci_runner.clean_up()
+        system_test_preparer.run()
+    except Exception as exc:
+        logger.error("Failed running system test automation", exc=exc)
+        raise
+    # finally:
+    #     system_test_preparer.clean_up_remote_workdir()
```

Comment:
why commented out ?

Reply:
needs to be removed.
After my slight changes to the provctl commands the workdir now holds logs for the 2 provctl commands used.
In case of failures it's very nice having this saved on the system after the run. This workdir is cleaned first thing when `run()` is called (so cleanup on set-up only, instead of both set-up and teardown)